test/ZoraNFTCreatorV1_Fork.t.sol

```diff
@@ -178,4 +189,45 @@ contract ZoraNFTCreatorV1Test is Test, ForkHelper {
         drop.purchase{value: fee}(1);
         assertEq(drop.tokenURI(1), "DEMO");
     }
+
+    function test_ForkUpgradeWithDifferentContractName() external {
+        string[] memory forkTestChains = getForkTestChains();
+
+        for (uint256 i = 0; i < forkTestChains.length; i++) {
+            string memory chainName = forkTestChains[i];
+            vm.createSelectFork(vm.rpcUrl(chainName));
+            creator = ZoraNFTCreatorV1(getDeployment().factory);
+            verifyAddressesFork(chainName);
+            deployCore();
+            forkContractNameWithRevert();
+            forkContractNameSucceed();
+        }
+    }
+
+    function forkContractNameWithRevert() internal {
+        ZoraNFTCreatorV1 newCreatorImpl = new ZoraNFTCreatorV1(address(dropImpl), editionMetadataRenderer, dropMetadataRenderer);
```

Comment:
not sure how this exactly takes advantage of the fact this is a fork test.  couldn't this just be done without fork testing? 

another approach could be to:

select a fork on a specific chain on a specific block #.  load an existing erc721 contract that we know has been deployed already by that block #.  attempt to upgrade it to a bad contract, it should revert

Reply:
The current contracts on chain do not have the contract name check in `_authorizeUpgrade` so it will not check for miss matched contract names. In order to get around that we upgrade first and then check

Reply:
ahh got it - i missed above how `creator` is assigned to from the fork address.

Reply:
for test readability, it may be better to add a comment that explains were connected to the forked contract in the test, or have a different variable name (like `forkedCreator`)

Reply:
okay will do!